scripts/ci_checks/linux-checks/check-version.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+
+tagname="$(git tag -l | tail -n1)"
+
+header_tagname="v$(python3 scripts/scons_helpers/parse-version.py)"
+
+if [[ "$tagname" != "$header_tagname" ]]; then
+  echo
+  echo "version check FAILED" >&2
+  echo "the git tag name and version declared in version.h is not the same" >&2
+  echo "${tagname} git version tag" >&2
+  echo "${header_tagname} header version tag" >&2
+  echo -e "\n" 
```

Comment:
This variant is more readable (and also prevents you from forgetting to do stream redirection as you did in lines 11 and 16):

```suggestion
  echo >&2
  echo >&2 "version check FAILED"
  echo >&2 "the git tag name and version declared in version.h is not the same"
  echo >&2 "${tagname} git version tag"
  echo >&2 "${header_tagname} header version tag"
  echo >&2 
```

Also `echo` without arguments prints a new line. `echo -e "\n"` prints two, but I suppose one is sufficient here.